spring-pulsar/src/main/java/org/springframework/pulsar/annotation/ReactivePulsarListener.java

```diff
@@ -0,0 +1,156 @@
+ * {@link ReactivePulsarListenerContainerFactory} to use to build the Pulsar listener
+ * container. If not set, a <em>default</em> container factory is assumed to be available
+ * with a bean name of {@code pulsarListenerContainerFactory} unless an explicit default
+ * has been provided through configuration.
+ *
+ * <p>
+ * Processing of {@code @ReactivePulsarListener} annotations is performed by registering a
+ * {@link ReactivePulsarListenerAnnotationBeanPostProcessor}. This can be done manually
+ * or, more conveniently, through {@link EnablePulsar} annotation.
+ * </p>
+ *
+ * @author Christophe Bornet
+ */
+@Target({ ElementType.TYPE, ElementType.METHOD, ElementType.ANNOTATION_TYPE })
+@Retention(RetentionPolicy.RUNTIME)
+@MessageMapping
+@Documented
+public @interface ReactivePulsarListener {
+
+	/**
+	 * The unique identifier of the container for this listener.
+	 * <p>
+	 * If none is specified an auto-generated id is used.
+	 * <p>
+	 * SpEL {@code #{...}} and property place holders {@code ${...}} are supported.
+	 * @return the {@code id} for the container managing for this endpoint.
+	 * @see ReactivePulsarListenerEndpointRegistry#getListenerContainer(String)
+	 */
+	String id() default "";
+
+	/**
+	 * Pulsar subscription name associated with this listener.
+	 * @return the {@code subscriptionName} for this Pulsar listener endpoint.
+	 */
+	String subscriptionName() default "";
+
+	/**
+	 * Pulsar subscription type for this listener.
+	 * @return the {@code subscriptionType} for this listener
+	 */
+	SubscriptionType subscriptionType() default SubscriptionType.Exclusive;
+
+	SchemaType schemaType() default SchemaType.NONE;
+
+	/**
+	 * The bean name of the {@link ReactivePulsarListenerContainerFactory} to use to
+	 * create the message listener container responsible to serve this endpoint.
+	 * <p>
+	 * If not specified, the default container factory is used, if any. If a SpEL
+	 * expression is provided ({@code #{...}}), the expression can either evaluate to a
+	 * container factory instance or a bean name.
+	 * @return the container factory bean name.
+	 */
+	String containerFactory() default "";
+
+	/**
+	 * Topics to listen to.
+	 * @return a comma separated list of topics to listen from.
+	 */
+	String[] topics() default {};
+
+	/**
+	 * Topic patten to listen to.
+	 * @return topic pattern to listen to.
+	 */
+	String topicPattern() default "";
+
+	/**
+	 * Set to true or false, to override the default setting in the container factory. May
+	 * be a property placeholder or SpEL expression that evaluates to a {@link Boolean} or
+	 * a {@link String}, in which case the {@link Boolean#parseBoolean(String)} is used to
+	 * obtain the value.
+	 * <p>
+	 * SpEL {@code #{...}} and property place holders {@code ${...}} are supported.
+	 * @return true to auto start, false to not auto start.
+	 */
+	String autoStartup() default "";
+
+	/**
+	 * Activate stream consumption.
+	 * @return whether this listener is a streaming one.
```

Comment:
In layman's terms, what is the result if this is `false`? Or put another way, what does a "streaming one" mean?

Reply:
OK. I guess the same can be said for the `PulsarListener::batch` javadoc ?

Reply:
Done